src/AerospikeCache.php

```diff
@@ -69,25 +66,28 @@ protected function doClear($namespace = ''): bool
     {
         if ($namespace === '') {
             $statusCode = $this->aerospike->truncate($this->namespace, $this->set, 0);
+            $cleared = $this->isStatusOkOrNotFound($statusCode);
         } else {
-            $clearNamespace = function ($record) use ($namespace): void {
+            $removedAllRecords = true;
+            $clearNamespace = function ($record) use ($namespace, &$removedAllRecords): bool {
                 if ($namespace === mb_substr($record['key']['key'], 0, mb_strlen($namespace))) {
-                    $status = $this->aerospike->remove($record['key']);
+                    $statusCodeFromRemove = $this->aerospike->remove($record['key']);
 
-                    if (!$this->isStatusOkOrNotFound($status)) {
-                        throw new CacheException();
+                    if (!$this->isStatusOkOrNotFound($statusCodeFromRemove)) {
+                        $removedAllRecords = false;
+
+                        return false;
```

Comment:
Changed the callback to void. There is no need to return true as it will do something (halt) only if false is returned. 
Now it will only keep track if the namespace was cleared completely.